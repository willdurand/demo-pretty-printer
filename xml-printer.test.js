const { parse } = require("@xml-tools/parser");
const { buildAst } = require("@xml-tools/ast");

const { renderDocument } = require("./document");
const { makeDocument } = require("./xml-printer");

describe(__filename, () => {
  const render = (xml, options) => {
    const { cst, tokenVector } = parse(xml);

    const doc = makeDocument(buildAst(cst, tokenVector), {
      originalText: xml,
    });

    return renderDocument(doc, options);
  };

  it("formats attributes", () => {
    const xml = '<node id="123" attr1="value1"></node>';

    expect(render(xml, { width: 50 })).toMatchSnapshot();
    expect(render(xml, { width: 10 })).toMatchSnapshot();
  });

  it("formats long content", () => {
    const xml = [
      "<node>Lorem ipsum dolor sit amet, consectetur adipiscing",
      "elit, sed do eiusmod tempor incididunt ut labore et dolore",
      "magna aliqua.</node>",
    ].join(" ");

    expect(render(xml, { width: 50 })).toMatchSnapshot();
    expect(render(xml, { width: 10 })).toMatchSnapshot();
  });
});
