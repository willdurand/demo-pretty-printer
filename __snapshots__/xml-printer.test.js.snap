// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/william/projects/demo-pretty-printer/xml-printer.test.js formats attributes 1`] = `"<node id=\\"123\\" attr1=\\"value1\\"></node>"`;

exports[`/Users/william/projects/demo-pretty-printer/xml-printer.test.js formats attributes 2`] = `
"<node
  id=\\"123\\"
  attr1=\\"value1\\"
>
  
</node>"
`;

exports[`/Users/william/projects/demo-pretty-printer/xml-printer.test.js formats long content 1`] = `
"<node>
  Lorem ipsum dolor sit amet, consectetur adipiscing
  elit, sed do eiusmod tempor incididunt ut labore
  et dolore magna aliqua.
</node>"
`;

exports[`/Users/william/projects/demo-pretty-printer/xml-printer.test.js formats long content 2`] = `
"<node>
  Lorem ipsum
  dolor sit
  amet, consectetur
  adipiscing
  elit, sed
  do eiusmod
  tempor incididunt
  ut labore
  et dolore
  magna aliqua.
</node>"
`;
